src/components/tableview/TableTable.vue

```diff
@@ -486,51 +566,142 @@ return dt.split("(")[0]
         cell: cell,
         value: cell.getValue(0)
       }
-      this.$set(this.pendingEdits, key, payload)
+
+      this.addPendingChange(CHANGE_TYPE_UPDATE, payload)
     },
-    async saveChanges() {
-      try {
-        // throw new Error("This is an error")
-        const newData = await this.connection.updateValues(this.pendingEditList)
-        const updateIncludedPK = this.pendingEditList.find(e => e.column === e.pkColumn)
-        if (updateIncludedPK) {
-          this.tabulator.replaceData()
-        } else {
-          this.tabulator.updateData(newData)
-          this.pendingEditList.forEach(edit => {
-            edit.cell.getElement().classList.remove('edited')
-            edit.cell.getElement().classList.add('edit-success')
-            setTimeout(() => {
-              edit.cell.getElement().classList.remove('edit-success')
-            }, 1000)
-          })
+    addRowToPendingInserts(row) {
+      row.getElement().classList.add('inserted')
+
+      const payload = {
+        table: this.table.name,
+        row: row,
+        schema: this.table.schema,
+        pkColumn: this.primaryKey
+      }
+
+      this.addPendingChange(CHANGE_TYPE_INSERT, payload)
+    },
+    addRowToPendingDeletes(row) {
+      const pkCell = row.getCells().find(c => c.getField() === this.primaryKey)
+
+      if (!pkCell) {
+        this.$noty.error("Can't delete row -- couldn't figure out primary key")       
+        return
+      }
+
+      if (this.hasPendingInserts && _.find(this.pendingChanges.inserts, { row: row })) {
+        this.$set(this.pendingChanges, 'inserts', _.reject(this.pendingChanges.inserts, { row: row }))
+        this.tabulator.deleteRow(row)
+        return
+      }
+
+      row.getElement().classList.add('deleted')
+
+      const payload = {
+        table: this.table.name,
+        row: row,
+        schema: this.table.schema,
+        pkColumn: this.primaryKey,
+        primaryKey: pkCell.getValue()
+      }
+
+      this.addPendingChange(CHANGE_TYPE_DELETE, payload)
+    },
+    addPendingChange(changeType, payload) {
+      if (changeType === CHANGE_TYPE_INSERT) {
+        this.pendingChanges.inserts.push(payload)
+      }
+
+      if (changeType === CHANGE_TYPE_UPDATE) {
+        // remove existing pending updates with identical pKey-column combo
+        let pendingUpdates = _.reject(this.pendingChanges.updates, { 'key': payload.key })
```

Comment:
const

Reply:
`pendingUpdates` is being modified on the next line.